### PR TITLE
Sim 2029 redshifting

### PR DESCRIPTION
**Pull request by rbiswas4**
This is a PR for https://jira.lsstcorp.org/browse/SIM-2029. 

Previous behavior for SNObject is preserved, but a new method has been added as a feature. The feature method is a method called 
```
def redshift(self, z, cosmo)
```
which can be used to change the redshift to z for a certain cosmology. 

- This is illustrated in `examples/SNObject_example.py`
- Tests have been added to `examples/testSN.py` to perform sanity checks
- Jenkins builds pass : https://ci.lsst.codes/job/stack-os-matrix/12321/

### CODE REVIEW

File: python/lsst/sims/catUtils/supernovae/snObject.py [hunk 1]
```diff
@@ -420,6 +419,49 @@ def mwEBVfromMaps(self):
...
+        Returns
+        -------
+        None, but it changes the instance
+
+        """
+        import types
+
+        # Check that the input redshift is a scalar
+        try:
+            assert isinstance(z, types.FloatType) or \
```
Comment (danielsf) [inline python/lsst/sims/catUtils/supernovae/snObject.py]: something I learned last week:  you can
```
import numbers
isinstance(x, numbers.Number)
```
to test if x is an int or a float

Comment (rbiswas4) [inline python/lsst/sims/catUtils/supernovae/snObject.py]: That would definitely be a better one. I have used this before but was having a hard time finding the exact syntax to do it.

**Comment by danielsf** [comment]
You asked me to take a look at this.  I think it looks fine.  If @jbkalmbach doesn't have time to look at it, I would say that it is ready to merge.

**Comment by jbkalmbach** [comment]
Looks good. Only suggestion is that in your example you show the relevant magnitudes changing/not changing as appropriate since those are the actual observables.

**Pull request merged**